biz.aQute.resolve/src/biz/aQute/resolve/AbstractResolveContext.java

```
@@ -135,7 +135,7 @@ protected synchronized void init() {
 	public List<Capability> findProviders(Requirement requirement) {
 		init();
 		List<Capability> result = findProviders0(requirement);
-		if (result == null || result.isEmpty()) {
+		if (result.isEmpty()) {
```

This is good! If findProviders0 ever returns null, we will fail fast.